realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -733,6 +819,16 @@ public SyncConfiguration build() {
                 throw new IllegalStateException("Could not create directory for saving the Realm: " + realmFileDirectory);
             }
 
+            if (!TextUtils.isEmpty(serverCertificateAssetName)) {
+                if (!syncClientValidateSsl) {
+                    throw new IllegalStateException("SSL Verification is disable, server certificate should not be provided");
```

Comment:
It feels a bit weird to me to have this check as well. The code for this would look like:

```
if (BuildType.DEBUG) {
  builder.disableSSLVerification()
} else {
  builder.trustedCA("myserver.pem")
}
```
Without this check it would be 
```
builder.trustedCA("myserver.pem")
if (BuildType.DEBUG) {
  builder.disableSSLVerification()
}
```
Which feels nicer to me?